### PR TITLE
Add basic PHP support

### PR DESCRIPTION
**Pull request by skaasten**
This will infer a PHP project using composer for dependencies and phpunit for tests, which are the defaults for Laravel projects.

I expect the test job won't pass for most projects, as other dependencies like a database or redis are often required. This is the minimum to get tests running.

### CODE REVIEW

File: generation/internal/php.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	jobs := []*Job{}
+
+	if hasLib(ls, "phpunit/phpunit") {
+		jobs = append(jobs, phpunitJob(ls))
+	}
+	return jobs
+}
+
+func hasLib(ls labels.LabelSet, lib string) bool {
```
Comment (polymeris) [inline generation/internal/php.go:21]: Maybe `hasPhpLib`?

File: generation/internal/php.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	})
+	return &Job{
+		Job: config.Job{
+			Name:         "test-php",
+			Comment:      "Install php packages and run tests",
+			Steps:        steps,
+			DockerImages: []string{"cimg/php:8.2.7-node"},
+		},
+		Orbs: map[string]string{
+			"php": "circleci/php@1.1.0",
```
Comment (polymeris) [inline generation/internal/php.go:52]: The orb seems to assume PHP 7.x: https://github.com/CircleCI-Public/php-orb/blob/master/src/commands/install-php.yml#L6
Is there potential for a compatibility issue with the php 8.x image?

It seems like PHP version can be specified in the `composer.json` file: https://getcomposer.org/doc/04-schema.md#package-links

File: generation/internal/php.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+func initialPhpSteps(ls labels.LabelSet) []config.Step {
+	checkout := checkoutStep(ls[labels.DepsPhp])
+	installPackages := config.Step{
+		Type:    config.OrbCommand,
+		Command: "php/install-packages",
+	}
+	return []config.Step{checkout, installPackages}
+}
+
+func phpunitJob(ls labels.LabelSet) *Job {
```
Comment (polymeris) [inline generation/internal/php.go:37]: `phpTestJob`?

**Review by polymeris — APPROVED** [review]
Some minor comments.

File: generation/internal/php.go (hunk 2 above)
Comment (skaasten) [reply generation/internal/php.go:52]: I think the orb commands we are using to install packages should work even in a later image. Determining the correct php version will be a challenge, particularly for repos that don't put the version in the composer.json. 

I'm going to leave looking looking for the version in composer.json for a future revision.

**Pull request merged**